modules/cli/apply/apply.go

```diff
@@ -110,16 +106,18 @@ func Apply(ctx context.Context, client *client.Getter, name string, baseDirs []s
 	return nil
 }
 
-func readBundle(ctx context.Context, name, baseDir string, opts *Options) (*bundle.Bundle, error) {
-	if opts.BundleReader != nil {
```

Comment:
No the reader is set in the calling function here

https://github.com/rancher/fleet/blob/450e23591526649cd8d388b2375ea590a1fbfff1/modules/cli/cmds/apply.go#L92-L114

This read bundle is just reading in the bundle from the given source. The issue stemmed that this only understands JSON so if the `fleet.yml` file was in YAML as it should be, it would fail to parse the file.

Reply:
I see... let me think about it